src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -523,11 +524,12 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
             }
         }
 
-        private void PrepareAuthorityInstanceForMsal()
+        private void PrepareAuthorityInstanceForMsal(string instance)
         {
-            if (_microsoftIdentityOptions.IsB2C && _applicationOptions.Instance.EndsWith("/tfp/"))
+            // we only use the cca options instance in this class
+            if (_microsoftIdentityOptions.IsB2C && instance.EndsWith("/tfp/"))
             {
-                _applicationOptions.Instance = _applicationOptions.Instance.Replace("/tfp/", string.Empty).Trim();
+                _applicationOptions.Instance = instance.Replace("/tfp/", string.Empty).Trim();
             }
 
             _applicationOptions.Instance = _applicationOptions.Instance.TrimEnd('/') + "/";
```

Comment:
Should this be updated?
```suggestion
            _applicationOptions.Instance = instance.TrimEnd('/') + "/";
```